### PR TITLE
Added new tests for utils

### PR DESCRIPTION
**Pull request by sauravk90**
Added test for newsunpack

### CODE REVIEW

File: tests/test_newsunpack.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import subprocess
+from sabnzbd.newsunpack import *
+from tests.testhelper import *
+
+
+class TestNewsUnpack():
+    def test_list_to_cmd(self):
+        """ Test to convert list to a cmd.exe-compatible command string """
+        lst = ['cmd1', 9, 'cmd3']
+        res = list2cmdline(lst)
```
Comment (Safihre) [inline tests/test_newsunpack.py]: Should also test what happens with `None`, `0` and `""`,

File: tests/test_utils/test_cert_gen.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+from sabnzbd.utils.certgen import generate_key, generate_local_cert
+
+
+class TestCertGen:
+    def test_generate_key(self):
+        # Generate private key
+        private_key = generate_key()
+
+        assert private_key.key_size == 2048
```
Comment (Safihre) [inline tests/test_utils/test_cert_gen.py]: Should also check if the output file is written to disk at `outfile` location.
Should also check if the `key_size` of the result changes if specified differently when calling the function.

File: tests/test_utils/test_cert_gen.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    def test_generate_local_cert(self):
+        # Generate private key
+        private_key = generate_key()
+        # Generate local certificate using private key
+        local_cert = generate_local_cert(private_key)
```
Comment (Safihre) [inline tests/test_utils/test_cert_gen.py]: Should also check the actual file.
Should do some checks on the resulting certificate, like expiration date and SAN-list.

File: tests/test_utils/test_diskspeed.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            dir = "/media/sander/"
+
+        elif "win32" in sys.platform:
+            dir = "D:/"
+
+        elif "darwin" in sys.platform:
+            dir = "/Volumes/CARTUNES/"
+
+        speed = diskspeedmeasure(dir)
+        # TODO : Provide a valid directory
```
Comment (Safihre) [inline tests/test_utils/test_diskspeed.py]: You can just use 
```python
import os
cwd = os.getcwd()
```
Directory doesn't really matter, since it should just be able to run anywhere.
After the test you should check if the `outputTESTING.txt` file is not present in the directory.

File: tests/test_utils/test_internetspeed.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+class TestInternetSpeed:
+    """ This class contains tests to measure internet speed with an active and inactive connection """
+    def test_measurespeed_invalid_url(self):
+        speed = measurespeed("www.fake-url-9999999.xyz")
+
+        assert isinstance(speed, float)
+        assert speed == 0.0
+
+    def test_measurespeed_valid_url(self):
+        speed = measurespeed("https://sabnzbd.org/tests/internetspeed/5MB.bin")
```
Comment (Safihre) [inline tests/test_utils/test_internetspeed.py]: Maybe use 
```python
from sabnzbd.utils.internetspeed import internetspeed, measurespeed, SizeUrlList
speed = measurespeed(SizeUrlList[0])
```
this way the URL only has to be changed in 1 location.

Comment (sauravk90) [reply tests/test_utils/test_internetspeed.py]: thanks, will accommodate all the above comments.

File: tests/test_newsunpack.py (hunk 1 above)
Comment (sauravk90) [reply tests/test_newsunpack.py]: Added more scenarios (parametrized). Also, one of the test with input as `["", "cmd1", "5"]` fails. I think there must be a check for `len(arg)>0 `in `list2cmdline`.

File: tests/test_utils/test_cert_gen.py (hunk 3 above)
Comment (sauravk90) [reply tests/test_utils/test_cert_gen.py]: Certificate validation added.

File: tests/test_utils/test_diskspeed.py (hunk 4 above)
Comment (sauravk90) [reply tests/test_utils/test_diskspeed.py]: Changes done.

File: tests/test_utils/test_internetspeed.py (hunk 5 above)
Comment (sauravk90) [reply tests/test_utils/test_internetspeed.py]: Thanks for the suggestion, I have used the `SizeUrlList` variable for test now.

File: tests/test_utils/test_cert_gen.py (hunk 2 above)
Comment (sauravk90) [reply tests/test_utils/test_cert_gen.py]: Added new test with multiple scenarios.
